src/main/java/org/jitsi/utils/logging2/JitsiLogFormatter.java

```diff
@@ -0,0 +1,243 @@
+        if (record.getThrown() != null)
+        {
+            try
+            {
+                StringWriter sw = new StringWriter();
+                PrintWriter pw = new PrintWriter(sw);
+                record.getThrown().printStackTrace(pw);
+                pw.close();
+                sb.append(sw.toString());
+            }
+            catch (Exception ex)
+            {
+            }
+        }
+        return sb.toString();
+    }
+
+    /**
+     * Try to extract the name of the class and method that called the current
+     * log statement.
+     *
+     * @param record the logrecord where class and method name should be stored.
+     *
+     * @return the line number that the call was made from in the caller.
+     */
+    private int inferCaller(LogRecord record)
+    {
+        // Get the stack trace.
+        StackTraceElement stack[] = (new Throwable()).getStackTrace();
+
+        //the line number that the caller made the call from
+        int lineNumber = -1;
+
+        // First, search back to a method in the SIP Communicator Logger class.
+        int ix = 0;
+        while (ix < stack.length)
+        {
+            StackTraceElement frame = stack[ix];
+            String cname = frame.getClassName();
+            if (cname.equals("org.jitsi.utils.logging2.LoggerImpl") ||
+                cname.equals("org.jitsi.utils.logging.LoggerImpl"))
+            {
+                break;
+            }
+            ix++;
+        }
+        // Now search for the first frame
+        // before the SIP Communicator Logger class.
+        while (ix < stack.length)
+        {
+            StackTraceElement frame = stack[ix];
+            lineNumber=stack[ix].getLineNumber();
+            String cname = frame.getClassName();
+            String shortName = cname.substring(cname.lastIndexOf(".") + 1);
+            if (!cname.contains("org.jitsi.utils.logging"))
+            {
+                // We've found the relevant frame.
+                record.setSourceClassName(shortName);
+                record.setSourceMethodName(frame.getMethodName());
+                break;
+            }
+            ix++;
+        }
+
+        return lineNumber;
+    }
+
+    /**
+     * Loads all config properties.
+     */
+    private void loadConfigProperties()
+    {
+        loadProgramNameProperty();
+        loadTimestampDisabledProperty();
+    }
+
+    /**
+     * Checks and loads timestamp disabled property if any.
+     */
+    private static void loadTimestampDisabledProperty()
+    {
+        LogManager manager = LogManager.getLogManager();
+        String cname = JitsiLogFormatter.class.getName();
+        timestampDisabled = Boolean.parseBoolean(
+                manager.getProperty(cname + DISABLE_TIMESTAMP_PROPERTY));
+    }
+
+    /**
+     * Load the programname property to be used in logs to identify Jitsi-based
+     * application which produced the logs
+     */
+    private static void loadProgramNameProperty()
```

Comment:
Do we still need the program name? I would like to unset it in more cases for the bridge because it is (slightly) easier to read logs without it.